File: Tes_Sprint_Asia/app_test_todolist/client/src/models/task.ts
import { SubTask } from "./subTask";

export class Task {
  constructor(
    public id: number,
    public description: string,
    public status: string,
    public subTasks: SubTask[] = [],
    public deadline: string
  ) {}
}

export class TaskRequest {
  constructor(
    public description?: string,
    public status?: string,
    public deadline?: string,
    public subTasks?: object[]
  ) {}
}

File: Tes_Sprint_Asia/app_test_todolist/server/app/models/task.go
package models

import "time"

type Task struct {
	Base

	Deadline *time.Time

	SubTasks []SubTask `json:"subTasks,omitempty"`
}

type TaskRequest struct {
	Description *string
	Status      *string
	Deadline    *string

	SubTask []SubTask `json:"subTasks,omitempty"`
}
